hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -299,6 +307,15 @@ def _get_local_constants() -> Constants:
     return _local_constants
 
 
+@contextmanager
+def with_register_backend(name, provider_cls):
+    try:
+        AVAILABLE_PROVIDERS[name] = provider_cls
+        yield
+    finally:
+        del AVAILABLE_PROVIDERS[name]
+
+
```

Comment:
I'm naming this with the assumption that we'll migrate `AVAILABLE_PROVIDERS` to `def register_backend(name, class_or_string)` at some point